net/mptcp/mptcp_blest.c

```diff
@@ -0,0 +1,487 @@
+// SPDX-License-Identifier: GPL-2.0
+/*	MPTCP Scheduler to reduce HoL-blocking and spurious retransmissions.
+ *
+ *	Algorithm Design:
+ *	Simone Ferlin <ferlin@simula.no>
+ *	Ozgu Alay <ozgu@simula.no>
+ *	Olivier Mehani <olivier.mehani@nicta.com.au>
+ *	Roksana Boreli <roksana.boreli@nicta.com.au>
+ *
+ *	Initial Implementation:
+ *	Simone Ferlin <ferlin@simula.no>
+ *
+ *	Additional Authors:
+ *	Daniel Weber <weberd@cs.uni-bonn.de>
+ *
+ *	This program is free software; you can redistribute it and/or
+ *	modify it under the terms of the GNU General Public License
+ *	as published by the Free Software Foundation; either version
+ *	2 of the License, or (at your option) any later version.
+ */
+
+#include <linux/module.h>
+#include <net/mptcp.h>
+
+static unsigned char optim __read_mostly = 1;
+module_param(optim, byte, 0644);
+MODULE_PARM_DESC(optim, "1 = retransmit, 2 = retransmit+penalize, 0 = off");
+
+static unsigned char lambda __read_mostly = 12;
+module_param(lambda, byte, 0644);
+MODULE_PARM_DESC(lambda, "Divided by 10 for scaling factor of fast flow rate estimation");
+
+static unsigned char max_lambda __read_mostly = 13;
+module_param(max_lambda, byte, 0644);
+MODULE_PARM_DESC(max_lambda, "Divided by 10 for maximum scaling factor of fast flow rate estimation");
+
+static unsigned char min_lambda __read_mostly = 10;
+module_param(min_lambda, byte, 0644);
+MODULE_PARM_DESC(min_lambda, "Divided by 10 for minimum scaling factor of fast flow rate estimation");
+
+static unsigned char dyn_lambda_good = 10; /* 1% */
+module_param(dyn_lambda_good, byte, 0644);
+MODULE_PARM_DESC(dyn_lambda_good, "Decrease of lambda in positive case.");
+
+static unsigned char dyn_lambda_bad = 40; /* 4% */
+module_param(dyn_lambda_bad, byte, 0644);
+MODULE_PARM_DESC(dyn_lambda_bad, "Increase of lambda in negative case.");
+
+struct blestsched_priv {
+	u32 last_rbuf_opti;
+	u32 min_srtt;
+	u32 max_srtt;
```

Comment:
This change makes sense.

There are some other names that I am unhappy with. But if I compare the code of the other schedulers it is hard to find a matching and consistent naming scheme.

Can you give me an advice for naming
- `bsp` matches with `dsp` and `rsp` but not the redundant scheduler that calls this `sk_data`
- `metabsp` sounds wired and could be renamed to `bsc` or `bscb` to make clear it's the control block of BLEST
- should I use a underscore for `minsk`, `bestsk`, `besttp`, etc. since I have seen both ways of writing this like `meta_sk`?

Reply:
Yeah, the naming is not consistent. Something we should probably cleanup in a separate PR (if you want to be the one ;-))



> This change makes sense.
> 
> There are some other names that I am unhappy with. But if I compare the code of the other schedulers it is hard to find a matching and consistent naming scheme.
> 
> Can you give me an advice for naming
> 
> * `bsp` matches with `dsp` and `rsp` but not the redundant scheduler that calls this `sk_data`

I think that the redundant scheduler's naming should be fixed there.

Instead of bsp, you could also use blest_p. I actually think that dsp and rsp are not very explicit. Might be better to name them def_p, rr_p, ...

> * `metabsp` sounds wired and could be renamed to `bsc` or `bscb` to make clear it's the control block of BLEST

What about blest_cb ?

> * should I use a underscore for `minsk`, `bestsk`, `besttp`, etc. since I have seen both ways of writing this like `meta_sk`?

Not needed.


All of that being said, you don't need to spend time renaming things. The PR is good as it is. Let me know if you want to do one more iteration, or not.

Reply:
I think I'm done with the renaming. Should I just squash all commits into one and sign it?

Reply:
> Should I just squash all commits into one and sign it?

Thanks for having done that! :)